proxy/src/outbound.rs

```diff
@@ -80,13 +102,20 @@ where
         &mut self,
         key: &Self::Key,
     ) -> Result<Self::Service, Self::RouteError> {
-        let &(ref authority, protocol) = key;
-        debug!("building outbound {:?} client to {:?}", protocol, authority);
+        let &(ref dest, protocol) = key;
+        debug!("building outbound {:?} client to {:?}", protocol, dest);
 
-        let resolve = self.discovery.resolve(
-            authority,
-            self.bind.clone().with_protocol(protocol),
-        );
+        let resolve = match *dest {
+            Destination::LocalSvc(ref authority) => {
+                Discovery::LocalSvc(self.discovery.resolve(
+                    authority,
+                    self.bind.clone().with_protocol(protocol),
+                ))
+            },
+            Destination::External(addr) => {
+                Discovery::External(Some((addr, self.bind.clone().with_protocol(protocol))))
```

Comment:
Extending my previous comment, if we use the `Destination` service then we resolve a name to an IP address here, but if we use don't then we did the resolution earlier. This asymmetry is surprising.